src/main/sessionwindow/sessionwindow.ts

```diff
@@ -712,40 +791,50 @@ export class SessionWindow implements IDisposable {
           'Restarting server using the selected Python enviroment'
         );
 
-        const env = this._registry.addEnvironment(path);
-
-        if (!env) {
+        try {
+          this._registry.addEnvironment(path);
+        } catch (error) {
+          let message = `Error! Python environment at '${path}' is not compatible.`;
+          if (
+            error?.type === PythonEnvResolveErrorType.RequirementsNotSatisfied
+          ) {
+            const requirementInstallCmd = encodeURIComponent(
+              this._registry.getRequirementsPipInstallCommand()
+            );
+            message = `Error! Required Python packages not found in the selected environment. You can install using <copyable-span label="${requirementInstallCmd}" copied="${requirementInstallCmd}"></copyable-span> command in the selected environment.`;
```

Comment:
Is it possible to show the detected version here>

Reply:
it is possible. it is easier to just log them though.